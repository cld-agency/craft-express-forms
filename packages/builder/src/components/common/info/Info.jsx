import React from 'react';
import './info.styl';
import IconInfo from './info-circle-solid.svg';
import Tooltip from '@material-ui/core/Tooltip';
import translate from '@xf/builder/functions/translator';

const Info = ({ description }) =>
  description ? (
    <Tooltip title={translate(description)} placement="right">
      <span className="information-dot">
        <IconInfo />
      </span>
    </Tooltip>
  ) : null;

export default Info;
