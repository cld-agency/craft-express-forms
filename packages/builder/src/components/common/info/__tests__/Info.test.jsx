import React from 'react';
import { shallow } from 'enzyme';
import Info from '../Info';

describe('<Info />', () => {
  it('renders & displays correctly', () => {
    const wrapper = shallow(<Info description="test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('contains the right classname', () => {
    const wrapper = shallow(<Info description="test" />);
    expect(wrapper.find('.information-dot')).toHaveLength(1);
  });

  it('contains the description', () => {
    const wrapper = shallow(<Info description="Test description provided" />);
    expect(wrapper.prop('title')).toStrictEqual('Test description provided');
  });

  it('does not render if no description provided', () => {
    const wrapper = shallow(<Info />);
    expect(wrapper.isEmptyRender()).toBeTruthy();
  });
});
