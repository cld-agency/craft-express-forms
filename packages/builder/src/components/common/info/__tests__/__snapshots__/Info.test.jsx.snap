// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> renders & displays correctly 1`] = `
<WithStyles(ForwardRef(Tooltip))
  placement="right"
  title="test"
>
  <span
    className="information-dot"
  >
    <SVGElement />
  </span>
</WithStyles(ForwardRef(Tooltip))>
`;
